pkg/project/resource_project_repo.go

```diff
@@ -153,13 +154,36 @@ var deleteRepos = func(ctx context.Context, projectKey string, repoKeys []RepoKe
 var deleteRepo = func(ctx context.Context, projectKey string, repoKey RepoKey, m interface{}) error {
 	tflog.Debug(ctx, fmt.Sprintf("deleteRepo: %s", repoKey))
 
-	_, err := m.(*resty.Client).
+	type Error struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+	}
+
+	type ErrorResponse struct {
+		Errors []Error `json:"errors"`
+	}
+
+	var errorResp ErrorResponse
+
+	resp, err := m.(*resty.Client).
 		R().
 		AddRetryCondition(retryOnSpecificMsgBody("A timeout occurred")).
 		AddRetryCondition(retryOnSpecificMsgBody("Web server is down")).
 		AddRetryCondition(retryOnSpecificMsgBody("Web server is returning an unknown error")).
 		SetPathParam("repoKey", string(repoKey)).
+		SetError(&errorResp).
 		Delete(projectsUrl + "/_/attach/repositories/{repoKey}")
 
+	// Ignore 404 NOT_FOUND error when unassigning repo from project
+	// Possible that repo was deleted out-of-band from TF
+	if resp.StatusCode() == http.StatusNotFound && len(errorResp.Errors) > 0 {
+		for _, error := range errorResp.Errors {
```

Comment:
Does there exist a case where you can get a 404 with an error that isn't NOT_FOUND. I'm curious as to what can be in this Errors array, and if its okay to ignore everything that isn't the not found error if the error exists.

Reply:
In theory, there shouldn't any other errors for 404. But since this response is not documented and we don't have access to the source code, we don't have ways to know.